### DISCUSSION TITLE
Time for a new release?

### DISCUSSION THREAD

**Discussion by jpivarski** [scikit-hep/vector#351] [General]
The last release was [v1.0.0](https://github.com/scikit-hep/vector/releases/tag/v1.0.0) in February.

[Since then](https://github.com/scikit-hep/vector/compare/v1.0.0...8ca75f6), there have been a lot of pre-commit, CI, etc. auto-updates, but one substantive change: #348, which updates the minimum Numba version. It would be better to not have conflicting versions of Numba.

Oh, actually, a new release should probably wait for @agoose77's #347, which adds another long-requested feature: `ak.sum` on vectors would sum properly in all coordinate systems (instead of giving the wrong result in old versions of Awkward and an exception in recent versions of Awkward).

Soon after that, you might want to be thinking of version 1.1.0, since a dependency version threshold changed.

**Comment by jpivarski** [comment]
@Saransh-cpp and @henryiii, I think it would be good to do a 1.1.0 release of Vector now or soon. The two new features are:

  1. Uses Numba's `isclose` implementation, rather than overlaying one on top of Numba (which is dangerous, now that Numba defines it).
  2. Overloads `ak.sum` (and `ak.count` and `ak.count_nonzero`) for vectors, which is a long-requested feature.

Both of these require new minimal Numba and Awkward thresholds (and both are optional dependencies). Since the dependency thresholds have changed, I think that a new minor version would make more sense.

**Reply by Saransh-cpp** [reply]
Changelog updated. Should be good to go! I can make the release if there are no blockers.

**Reply by henryiii** [reply]
Let's switch to trusted publisher releases, I'm switching on next release for every package that gets released.

Did we decide to bump the required Python version? Don't remember.

**Reply by henryiii** [reply]
#354

**Reply by Saransh-cpp** [reply]
> Did we decide to bump the required Python version? Don't remember.

I'm not sure as well. Maybe, @jpivarski can confirm this. The required version right now is `3.7` (but `3.8` for `numba`).

**Reply by Saransh-cpp** [reply]
The new release is up - https://github.com/scikit-hep/vector/releases/tag/v1.1.0!

**Comment by ianna** [comment]
@Saransh-cpp and @henryiii - it would be good to make a new release of Vector soon. We'd like to benefit from performance improvements @pfackeldey has implemented. Is it the right place to ask for it? Thanks!

**Reply by Saransh-cpp** [reply]
I'd be happy to make one! Given that we have a new feature (`awkward_transform`), should it be `v1.6.0` or just `v1.5.3`?

**Reply by pfackeldey** [reply]
I'd argue it's a relevant improvement and new feature (even though expert-only) that it deserves `v1.6.0`, but then I'm also not an expert on SemVer. (This is based on my gut feeling.)

**Reply by henryiii** [reply]
Gut feeling >> SemVer. (SerVer is a fuzzy concept in practice, at least for languages like Python that allow users to touch all the internals of any library)

**Reply by Saransh-cpp** [reply]
Up here - https://pypi.org/project/vector/1.6.0/!
